app.rb

```diff
@@ -1,124 +1,110 @@
-      end
-    end
+require_relative 'book'
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+
+# List all books
+def list_books(books)
+  books.each do |book|
+    puts "Title: #{book.title}, Author: #{book.author}"
   end
+end
 
-  def create_person
-    puts 'Do you want to create a student (1) or a teacher(2)?....'
-    option = gets.chomp
-
-    case option
-    when '1'
-      create_student
-    when '2'
-      create_teacher
-    else
-      puts 'Invalid input! Please type 1 or 2'
+# List all people
+def list_people(people)
+  people.each do |person|
+    if person.is_a?(Teacher)
+      puts "[Teacher] Name:#{person.name}, ID:#{person.id}, Age:#{person.age}"
+    elsif person.is_a?(Student)
+      puts "[Student] Name:#{person.name}, ID:#{person.id}, Age:#{person.age}"
     end
   end
+end
 
-  def create_student
-    print 'Name:'
-    name = gets.chomp
-
-    print 'Age:'
-    age = gets.chomp.to_i
-
-    print 'Has Parent permission? [Y/N]:'
-    parent_permission = gets.chomp.downcase
-
-    student = Student.new(age, @classroom, name, parent_permission: parent_permission == 'y')
-    @people << student
-    puts 'Student created successfully'
-  end
-
-  def create_teacher
-    print 'Name:'
-    name = gets.chomp
-
-    print 'Age:'
-    age = gets.chomp.to_i
-
-    print 'Specialization:'
-    specialization = gets.chomp
-
-    @people << Teacher.new(age, specialization, name)
-    puts 'Teacher created successfully'
-  end
-
-  def create_book
-    print 'Title: '
-    title = gets.chomp
-
-    print 'Author: '
-    author = gets.chomp
-
-    @books << Book.new(title, author)
-
-    puts 'Book created successfully'
+# Create a person (teacher or student)
+def create_person(people)
+  puts 'Do you want to create a student(1) or a teacher(2)?[Input the number]: '
+  choice = gets.chomp.to_i
+  print 'Enter age: '
+  age = gets.chomp.to_i
+  print 'Enter name: '
+  name = gets.chomp
+  print 'Does the person have parent permission? (y/n): '
+  parent_permission = gets.chomp.downcase == 'y'
+
+  case choice
+  when 1
+    person = Student.new(people.length + 1, age, name, parent_permission)
+  when 2
+    create_teacher(people, age, name, parent_permission)
+    return
+  else
+    puts 'Invalid choice'
+    return
   end
 
-  def create_rental
-    puts 'Select a book from the following list by number'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
-    end
+  people << person
+  puts "Created person #{person.name}"
+end
 
-    book_id = gets.chomp.to_i
+def create_teacher(people, age, name, parent_permission)
+  print 'Enter specialization: '
+  specialization = gets.chomp
+  person = Teacher.new(people.length + 1, age, name, parent_permission, specialization)
+  people << person
+  puts "Created teacher #{person.name}"
+end
```

Comment:
- [ ] as the previous CR mentioned any code in relation to input and input validation should be placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. this includes all the `gets.chomp` parts of the code.